backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -1194,6 +1221,21 @@ private class InteropTransformer(val context: Context, override val irFile: IrFi
                         putValueArgument(3, jobPointer)
                     }
                 }
+                IntrinsicType.CREATE_CLEANER -> {
+                    val irCallableReference = expression.getValueArgument(1)
+                    if (irCallableReference == null || !irCallableReference.isNonCapturingFunction()) {
+                        context.reportCompilationError(
+                                "${function.fqNameForIrSerialization} must take an unbound, non-capturing function or lambda",
+                                irFile, expression
+                        )
+                    }
+
+                    builder.irCall(symbols.createCleanerImpl).apply {
```

Comment:
Because intrinsics are expected to be replaced. It's handled here:
https://github.com/JetBrains/kotlin-native/blob/c2d05acf710d6c88a768f747760c400bd3f417e2/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt#L2153
I can workaround it by introducing predicate whether intrinsic must be replaced or should be evaluated with its original expression, but do we need it?

Reply:
I guess the more straightforward solution would be just to make `createCleaner` a regular function and add a compiler check (which is going to be moved to a special pass soon, and then to the frontend).